aes-gcm-siv/src/lib.rs

```diff
@@ -301,8 +301,8 @@ where
         self.polyval.update_padded(associated_data);
 
         // TODO(tarcieri): interleave decryption and authentication
-        init_ctr(&self.enc_cipher, tag).apply_keystream_partial(buffer.into());
-        self.polyval.update_padded(buffer);
+        init_ctr(&self.enc_cipher, tag).apply_keystream_partial(buffer.reborrow());
+        self.polyval.update_padded(buffer.get_in());
```

Comment:
I think this one needs to be `get_out`, since it's authenticating the decrypted plaintext (since the "SIV" tag is calculated from the plaintext)

Reply:
Yes